classes/class-ledyer-confirmation.php

```diff
@@ -27,22 +27,55 @@ public function actions() {
 		add_action( 'init', array( $this, 'confirm_order' ), 10, 2 );
 	}
 
-
 	/**
 	 * Confirm the order in Woo
 	 */
 	public function confirm_order() {
-		$ledyer_confirm = filter_input( INPUT_GET, 'lco_confirm', FILTER_SANITIZE_URL);
-		$order_key    = filter_input( INPUT_GET, 'key', FILTER_SANITIZE_STRING);
+		$ledyer_confirm  = filter_input( INPUT_GET, 'lco_confirm', FILTER_SANITIZE_URL );
+		$order_key       = filter_input( INPUT_GET, 'key', FILTER_SANITIZE_SPECIAL_CHARS );
+		$ledyer_order_id = filter_input( INPUT_GET, 'ledyer_id', FILTER_SANITIZE_SPECIAL_CHARS );
 
 		if ( empty( $ledyer_confirm ) || empty( $order_key ) ) {
 			return;
 		}
 		$order_id = wc_get_order_id_by_order_key( $order_key );
+
 		if ( empty( $order_id ) ) {
+			\Ledyer\Logger::log( 'Could not get wc order id from key ' . $order_key );
+			return;
+		}
+
+		$order = wc_get_order( $order_id );
+		if ( empty( $order ) ) {
+			\Ledyer\Logger::log( 'Could not get wc order ' . $order_id );
 			return;
 		}
 
+		// Check if is HPP.
+		if ( ! empty( $ledyer_order_id ) ) {
+			$ledyer_order = ledyer()->api->get_order( $ledyer_order_id );
+			if ( is_wp_error( $ledyer_order ) ) {
+				\Ledyer\Logger::log( 'Could not get ledyer order ' . $ledyer_order_id );
```

Comment:
Change `ledyer` to `Ledyer`.

Reply:
Also specify that the string following the message is a Ledyer order id. So for example: `\Ledyer\Logger::log( 'Could not get the order from Ledyer with the id: ' . $ledyer_order_id );`